exchange/bitswap/bitswap.go

```diff
@@ -208,11 +208,14 @@ func (bs *Bitswap) GetBlocks(ctx context.Context, keys []key.Key) (<-chan *block
 		log.Event(ctx, "Bitswap.GetBlockRequest.Start", &k)
 	}
 
-	bs.wm.WantBlocks(keys)
+	bs.wm.WantBlocks(ctx, keys)
 
-	req := &blockRequest{
-		keys: keys,
-		ctx:  ctx,
+	// NB: Optimization. Assumes that providers of key[0] are likely to
```

Comment:
Yeah, I thought broadcasting every blocks is too granular considering the bandwidth cost & rtt. This reduces dup subsequent blocks download as well?